api/views/yangSearch/grep_search.py

```diff
@@ -0,0 +1,127 @@
+import json
+import os
+import subprocess
+import typing as t
+from configparser import ConfigParser
+
+from api.views.yangSearch.response_row import ResponseRow
+from elasticsearchIndexing.es_manager import ESManager
+from elasticsearchIndexing.models.es_indices import ESIndices
+from redisConnections.redisConnection import RedisConnection
+from utility import log
+from utility.create_config import create_config
+from utility.util import validate_revision
+
+
+class GrepSearch:
+    """
+    Class is used to perform a grep-like search through text of all available modules.
+    """
+
+    def __init__(
+            self,
+            config: ConfigParser = create_config(),
+            es_manager: ESManager = ESManager(),
+            modules_es_index: ESIndices = ESIndices.YINDEX,
+            redis_connection: RedisConnection = RedisConnection(),
+    ):
+        all_modules_directory = config.get('Directory-Section', 'save-file-dir')
+        self.all_modules_directory = (
```

Comment:
not really, I can't just use ```.lstrip('/')``` here because this path has to begin with ```/```. moreover, I receive results like this: ```/var/yang/all_modules/iana-routing-types@2022-08-19.yang```, so it's important that ```self.all_modules_directory``` ends with ```/``` and I can just split the result string and retrieve the module name. for example, in the shell command itself, I don't need ```/``` at the end of ```self.all_modules_directory```, so I just use ```self.all_modules_directory.rstrip("/")``` there

Reply:
What I meant was instead of adding the `/` to the end when getting the value from the config, you could do
```python
module_name_with_format = result.split(self.all_modules_directory)[1].split(':')[0].lstrip('/')
```
The sample config file has all directory paths without the trailing `/`, and that's the way it is in most of the codebase, so if it can be done without issues that way here too, I think that would be better.

Reply:
for example, after using ```result.split(self.all_modules_directory)[1].split(':')[0]``` on ```/var/yang/all_modules/iana-routing-types@2022-08-19.yang``` I get ```iana-routing-types@2022-08-19.yang```, so ```.lstrip('/')``` wouldn't help here I think

Reply:
I don't really need ```.split(':')[0]``` here anymore because ```-l``` flag is used and I only receive results like this: ```/var/yang/all_modules/iana-routing-types@2022-08-19.yang```, so after using ```result.split(self.all_modules_directory)[1]``` I'll receive ```iana-routing-types@2022-08-19.yang```, as I can see, ```.lstrip('/')``` wouldn't help here

Reply:
Yes I understand that, what I'm trying to say is that you could remove
```python
        self.all_modules_directory = (
            all_modules_directory if
            all_modules_directory.endswith('/') else
            f'{all_modules_directory}/'
        )
```
Then you'd be getting `/iana-routing-types@2022-08-19.yang` after the split.

Reply:
oh, I got it, fixed, thanks